verilog/formatting/token_annotator.cc

```diff
@@ -206,8 +206,12 @@ static WithReason<int> SpacesRequiredBetween(
       return {0, "No space around streaming operators"};
     } else if (left.format_token_enum == FormatTokenType::numeric_literal ||
                left.format_token_enum == FormatTokenType::identifier ||
-               left.format_token_enum == FormatTokenType::keyword) {
+               (left.format_token_enum == FormatTokenType::keyword &&
+                strcmp(verilog_symbol_name(left.TokenEnum()), "\"return\""))) {
       return {0, "No space around streaming operator slice size"};
+    } else if (left.format_token_enum == FormatTokenType::keyword &&
+               !strcmp(verilog_symbol_name(left.TokenEnum()), "\"return\"")) {
+      return {1, "Space between return keyword and return value"};
```

Comment:
I would suggest instead of editing inside the condition for `kStreamingConcatenation` that we think of this as a matter of rule precedence, that is `return ...` should always have a space no matter what is on the right.  See the block around line 159 (if keyword).  It was intended to cover `return` but wasn't reached first.

* If it is possible/easy to reorder rules that don't break existing tests and fix the new ones great!  (What happens if this whole block were moved further down?)
* If that proves difficult, the next easiest option is to add one case like: `if (left.token_enum() == TK_return) ...`

Reply:
I tried your suggestion of re-arranging the blocks. I moved this block that I updated to the bottom - along with some other blocks to pass all the cases - but still, no spaces were added after the return keyword.
I believe adding this condition as you've suggested before the block is the easiest option:
```cpp
  if (left.TokenEnum() == TK_return){
    return {1, "Space between return keyword and return value"};
  }
```

Reply:
Ok, thanks for checking.  Let's go with this solution.